worlds/minecraft/Rules.py

```diff
@@ -7,270 +7,465 @@
 
 from . import Constants
 
+
 # Helper functions
 # moved from logicmixin
 
-def has_iron_ingots(state: CollectionState, player: int) -> bool:
+def has_iron_ingots(self, state: CollectionState, player: int) -> bool:
```

Comment:
Are the unused parameters (`self` everywhere except `combat_difficulty` which adds `state` and `player`) in these just being added to the formatting can be consistent?

Reply:
I know they aren't used it was mainly so everything stayed consistent, and you can actually reach your options if you need it easily for future changes.

Reply:
it would be nice in that case if it could be renamed & typed, since `self` without typing invokes some pretty strong associations & even gets colored differently by IDEs

Not strictly necessary for the code to run, but yea